analyzer/src/main/kotlin/managers/Pip.kt

```diff
@@ -206,23 +207,19 @@ class Pip(
     private fun runPipInVirtualEnv(virtualEnvDir: File, workingDir: File, vararg commandArgs: String) =
         runInVirtualEnv(virtualEnvDir, workingDir, command(workingDir), *TRUSTED_HOSTS, *commandArgs)
 
-    private fun runInVirtualEnv(virtualEnvDir: File, workingDir: File, commandName: String, vararg commandArgs: String):
-            ProcessCapture {
+    private fun runInVirtualEnv(
+        virtualEnvDir: File,
+        workingDir: File,
+        commandName: String,
+        vararg commandArgs: String
+    ): ProcessCapture {
         val binDir = if (Os.isWindows) "Scripts" else "bin"
-        var command = virtualEnvDir.resolve(binDir + File.separator + commandName)
-
-        if (Os.isWindows && command.extension.isEmpty()) {
-            // On Windows specifying the extension is optional, so try them in order.
-            val extensions = Os.env["PATHEXT"]?.splitToSequence(File.pathSeparatorChar).orEmpty()
-            val commandWin = extensions.map { File(command.path + it.toLowerCase()) }.find { it.isFile }
-            if (commandWin != null) {
-                command = commandWin
-            }
-        }
+        val command = virtualEnvDir.resolve(binDir).resolve(commandName)
+        val resolvedCommand = resolveWindowsExecutable(command)?.takeIf { Os.isWindows } ?: command
```

Comment:
It may be better to use an if-else statement here, as with the `takeIf()` construct `resolveWindowsExecutable()` is always executed.

Reply:
I had it with `if-else` before, which resulted in not-so-nice code due to the "double fallback" to `command`:

```
if (Os.isWindows) {
    resolveWindowsExecutable(command) ?: command
} else {
    command
}
```

So as calling `resolveWindowsExecutable()` is rather cheap, esp. on non-Windows, I decided to use the `takeIf()` approach for style 😎

Reply:
Maybe `command.takeUnless { it.isWindows } ...` would help?

Reply:
Nope, `command` is evaluated before `takeUnless` / `takeIf` in any case, unfortunately.